File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_map = {}
        for i in range(len(nums)):
            reqd = target - nums[i]
            if reqd in my_map:
                return [my_map[reqd], i]
            my_map[nums[i]]= i
        return []

File: Data Structures & Algorithms/target-sum/submission-1.cpp
class Solution {
public:
    int dpsolution(vector<int>& nums,int index, int target){
        if(index == nums.size()){
            return target == 0;
        }

        int addelement = dpsolution(nums, index + 1, target - nums[index]);
        int subtractelement = dpsolution(nums, index + 1, target + nums[index]);

        return addelement + subtractelement;
    }
    int findTargetSumWays(vector<int>& nums, int target) {
        int n = nums.size();
        return dpsolution(nums, 0, target);
    }
};
